frontend/src/metabase/components/DateMonthYearWidget/DateMonthYearWidget.tsx

```diff
@@ -8,9 +7,25 @@ import YearPicker from "metabase/components/YearPicker";
 
 import { MonthContainer, MonthList } from "./DateMonthYearWidget.styled";
 
-export default class DateMonthYearWidget extends React.Component {
-  constructor(props, context) {
-    super(props, context);
+type Props = {
+  value: string;
+  setValue: (v: string) => void;
+  onClose: () => void;
+};
+
+type State = {
+  month: number | null;
+  year: number;
+};
+
+class DateMonthYearWidget extends React.Component<Props> {
+  state: State = {
+    month: null,
+    year: moment().year(),
+  };
```

Comment:
I guess the preferred way to define the `state` type is to do `React.Component<Props, State>`

Reply:
Done!